sw/host/tests/manuf/manuf_sram_program_crc_check/src/main.rs

```diff
@@ -0,0 +1,123 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::time::Duration;
+
+use anyhow::{bail, Context, Result};
+use structopt::StructOpt;
+
+use opentitanlib::app::TransportWrapper;
+use opentitanlib::execute_test;
+use opentitanlib::io::jtag::JtagTap;
+use opentitanlib::test_utils::init::InitializeTest;
+use opentitanlib::test_utils::load_sram_program::{
+    execute_sram_program, ExecutionError, ExecutionMode, ExecutionResult, SramProgramParams,
+};
+use opentitanlib::uart::console::UartConsole;
+
+// use top_earlgrey::top_earlgrey_memory;
+
+#[derive(Debug, StructOpt)]
+struct Opts {
+    #[structopt(flatten)]
+    init: InitializeTest,
+
+    #[structopt(flatten)]
+    sram_program: SramProgramParams,
+}
+
+fn test_sram_load(opts: &Opts, transport: &TransportWrapper) -> Result<()> {
+    let uart = transport.uart("console")?;
+    //
+    // Connect to the RISC-V TAP
+    //
+    transport.pin_strapping("PINMUX_TAP_RISCV")?.apply()?;
+    transport.reset_target(opts.init.bootstrap.options.reset_delay, true)?;
+
+    let jtag = opts.init.jtag_params.create(transport)?;
+    log::info!("Connecting to RISC-V TAP");
+    jtag.connect(JtagTap::RiscvTap)?;
+    log::info!("Halting core");
+    jtag.halt()?;
+
+    // Load program on the device.
+    let sram_prog_info = opts.sram_program.load(&jtag)?;
+
+    // Execute and make sure it works (executes and print hello)
+    let exec_res = execute_sram_program(
+        &jtag,
+        &sram_prog_info,
+        ExecutionMode::JumpAndWait(Duration::from_secs(5)),
+    )?;
+    match exec_res {
+        ExecutionResult::ExecutionDone => {
+            log::info!("SRAM program finished successfully")
+        }
+        ExecutionResult::ExecutionError(err) => {
+            bail!("SRAM program execution failed with {:?}", err)
+        }
+        _ => {
+            bail!(
+                "SRAM program execution failed with unexpected result {:?}",
+                exec_res
+            )
+        }
+    }
+
+    const CONSOLE_TIMEOUT: Duration = Duration::from_secs(1);
+    let _ = UartConsole::wait_for(
+        &*uart,
+        r"sram_empty_functest\.c:\d+\] hello.",
+        CONSOLE_TIMEOUT,
+    )
+    .context("SRAM program did not print 'hello' in time")?;
+
+    // Modify a random word in the code, not too close to the starting point so that
```

Comment:
since there is a positive and negative test case (which is great!) maybe you could split them into two separate `execute_test!(...)` calls, so we get the benefit of the nice start/end console messages?

Reply:
done